crates/uv-settings/src/lib.rs

```diff
@@ -59,6 +59,20 @@ impl FilesystemOptions {
         }
     }
 
+    pub fn system() -> Result<Option<Self>, Error> {
+        if let Some(file) = system_config_file() {
+            match read_file(&file) {
```

Comment:
Now that the 3rd case (not a dir) is removed, you can just `let options = read_file(&file)?` I think.

See https://doc.rust-lang.org/beta/book/ch09-02-recoverable-errors-with-result.html#a-shortcut-for-propagating-errors-the--operator